### PR TITLE
Add initial support to produce images in ConfigMap as release artifact

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes

- Add initial support to produce images in ConfigMap as release artifact

/cc @matzew

### CODE REVIEW

File: hack/release.sh [hunk 1]
```diff
@@ -25,6 +28,38 @@ function build_release() {
   ./mvnw clean package -P knative -DskipTests || return $?
 
   echo "Image build & push completed"
+
+  echo "Collect image shas into ConfigMap"
+  generate_configMap
+
+  ARTIFACTS_TO_PUBLISH="${CONFIMAP_YAML} ${IMAGES_TXT}"
+  sha256sum ${ARTIFACTS_TO_PUBLISH} >checksums.txt
+  ARTIFACTS_TO_PUBLISH="${ARTIFACTS_TO_PUBLISH} checksums.txt"
```
Comment (dsimansk) [inline hack/release.sh:37]: `checksum.txt` is required by release script itself to complete successfully.

File: hack/release.sh [hunk 2]
```diff
@@ -25,6 +28,38 @@ function build_release() {
   ./mvnw clean package -P knative -DskipTests || return $?
 
   echo "Image build & push completed"
+
+  echo "Collect image shas into ConfigMap"
+  generate_configMap
+
+  ARTIFACTS_TO_PUBLISH="${CONFIMAP_YAML} ${IMAGES_TXT}"
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply hack/release.sh:35]:
```
https://storage.googleapis.com/knative-nightly/eventing-integrations/latest/eventing-integrations-images.yaml
```
Hopefully, but we need to test it in production. :)

[the remark this answers is not in the document]
Comment (dsimansk) [reply hack/release.sh:35]: Perhaps, dropping `eventing-integrations` from the filename to get rid off duplicated name. WDYT?

[the remark this answers is not in the document]
Comment (matzew) [reply hack/release.sh:35]: I like `eventing-integrations-images.yam`

I think that makes sense, when we require it to end up in https://github.com/knative/eventing/tree/main/config/core/configmaps

Or perhaps just `integration-images`?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/hold
In case we wanna do something on the filename

**Pull request merged**